addons/account_check_printing/models/account_payment.py

```diff
@@ -258,51 +258,74 @@ def prepare_vals(invoice, partials):
-
-        # Group partials by invoices.
-        invoice_map = {invoice: self.env['account.partial.reconcile'] for invoice in invoices}
-        for partial in term_lines.matched_debit_ids:
-            invoice = partial.debit_move_id.move_id
-            if invoice in invoice_map:
-                invoice_map[invoice] |= partial
-        for partial in term_lines.matched_credit_ids:
-            invoice = partial.credit_move_id.move_id
-            if invoice in invoice_map:
-                invoice_map[invoice] |= partial
-
-        # Prepare stub_lines.
-        if 'out_refund' in invoices.mapped('move_type'):
-            stub_lines = [{'header': True, 'name': "Bills"}]
-            stub_lines += [prepare_vals(invoice, partials)
-                           for invoice, partials in invoice_map.items()
-                           if invoice.move_type == 'in_invoice']
-            stub_lines += [{'header': True, 'name': "Refunds"}]
-            stub_lines += [prepare_vals(invoice, partials)
-                           for invoice, partials in invoice_map.items()
-                           if invoice.move_type == 'out_refund']
+        if self.move_id:
+            # Decode the reconciliation to keep only invoices.
+            term_lines = self.move_id.line_ids.filtered(lambda line: line.account_id.account_type in ('asset_receivable', 'liability_payable'))
+            invoices = (term_lines.matched_debit_ids.debit_move_id.move_id + term_lines.matched_credit_ids.credit_move_id.move_id)\
+                .filtered(lambda x: x.is_outbound() or x.move_type == 'in_receipt')
+
+            invoices = invoices.sorted(lambda x: x.invoice_date_due or x.date)
+
+            # Group partials by invoices.
+            invoice_map = {invoice: self.env['account.partial.reconcile'] for invoice in invoices}
+            for partial in term_lines.matched_debit_ids:
+                invoice = partial.debit_move_id.move_id
+                if invoice in invoice_map:
+                    invoice_map[invoice] |= partial
+            for partial in term_lines.matched_credit_ids:
+                invoice = partial.credit_move_id.move_id
+                if invoice in invoice_map:
+                    invoice_map[invoice] |= partial
+
+            # Prepare stub_lines.
+            if 'out_refund' in invoices.mapped('move_type'):
+                stub_lines = [{'header': True, 'name': "Bills"}]
+                stub_lines += [prepare_vals(invoice, partials=partials)
+                            for invoice, partials in invoice_map.items()
+                            if invoice.move_type == 'in_invoice']
+                stub_lines += [{'header': True, 'name': "Refunds"}]
+                stub_lines += [prepare_vals(invoice, partials=partials)
+                            for invoice, partials in invoice_map.items()
+                            if invoice.move_type == 'out_refund']
+            else:
+                stub_lines = [prepare_vals(invoice, partials)
```

Comment:
For the sake of consistency with a few lines above
```suggestion
                stub_lines = [prepare_vals(invoice, partials=partials)
```